exes/mqtt-broadcaster/inc/config.hpp

```diff
@@ -12,6 +12,7 @@ struct config {
   std::string node_id{};
   std::string group_id{};
   std::string client_id{};
+  std::array<std::tuple<std::string, std::string>, 10> scada_signals{};
```

Comment:
Couple of things I would like to note here,

1. std::array instead of std::vector?
2. std::tuple instead of struct? You could remove the following description `each signal is a tuple of (signal name, signal type). Allowed types are: _bool, _double_t, _int64_t, _json, _string, _uint64_t` by using a struct.

Example:
```cpp
struct signal_definition {
std::string name{};
tfc::ipc::details::type_e type{};
struct glaze {
// ...
};
};

// in mqtt config struct would be
struct config {
// ...
std::vector<signal_definition> scada_signals{};
};
```

Reply:
I agree std::vector should be used instead of std::array.

Initially I used a struct, similar to what you outlined above but backed off for the following reason.

Implementing the solution given above results in a compile error that originates in "json_schema.hpp". The compile error is:

```bash
error: no matching function for call to ‘get<2>(const glz::tuplet::tuple<std::basic_string_view<char, std::char_traits<char> >, tfc::ipc::details::type_e, std::basic_string_view<char, std::char_traits<char> > >&)’
  251 |         enumeration.attributes.description = std::get<2>(item);
```

This compile error happens when the configuration struct is initialized.

Reply:
This should be possible, so the compiler error just needs to be resolved

Reply:
After some debugging, the following is possible:

```cpp
struct signal_defintion {
  std::string name{};
  std::string type{};
  struct glaze {
    // clang-format off
        static constexpr auto value{ glz::object(
          "name", &signal_defintion::name,
          "type", &signal_defintion::type
        )};
    // clang-format on
  };
};
```

but the defining an internal tfc type for a variable is not possible:

```cpp
struct signal_defintion {
  std::string name{};
  tfc::ipc::details::type_e type{};
  struct glaze {
    // clang-format off
        static constexpr auto value{ glz::object(
          "name", &signal_defintion::name,
          "type", &signal_defintion::type
        )};
    // clang-format on
  };
};
```

Reply:
This now uses tfc types